presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2365,18 +2365,19 @@ public void createMaterializedView(ConnectorSession session, ConnectorTableMetad
         }
 
         Table basicTable = prepareTable(session, viewMetadata, MATERIALIZED_VIEW);
+        ConnectorMaterializedViewDefinition viewDefinitionWithRefreshColumns = viewDefinition.withValidRefreshColumns(getPartitionedBy(viewMetadata.getProperties()));
         Map<String, String> parameters = ImmutableMap.<String, String>builder()
                 .putAll(basicTable.getParameters())
                 .put(PRESTO_MATERIALIZED_VIEW_FLAG, "true")
                 .build();
         Table viewTable = Table.builder(basicTable)
                 .setParameters(parameters)
-                .setViewOriginalText(Optional.of(encodeMaterializedViewData(MATERIALIZED_VIEW_JSON_CODEC.toJson(viewDefinition))))
+                .setViewOriginalText(Optional.of(encodeMaterializedViewData(MATERIALIZED_VIEW_JSON_CODEC.toJson(viewDefinitionWithRefreshColumns))))
```

Comment:
No need to call it `viewDefinitionWithRefreshColumns`, just reuse `viewDefinition`